tests/driver/api.test.ts

```diff
@@ -0,0 +1,633 @@
+            const createResp = await Product.create({
+                name: 'Product for create',
+                price: 10,
+                isCertified: true,
+                category: 'cat 1'
+            });
+            assert.strictEqual(createResp.name, 'Product for create');
+        });
+        it('API ops tests Model.createCollection()', async () => {
+            await Product.createCollection();
+        });
+        //TODO - createIndex() error uncaught internally
+        it.skip('API ops tests Model.createIndexes()', async () => {
+            let error: any = null;
+            try {
+                Product.schema.index({name: 1});
+                await Product.createIndexes();
+            } catch (err: any) {
+                error = err;
+            }
+            assert.strictEqual(error!.message, 'createIndex() Not Implemented');
+        });
+        it('API ops tests Model.db', async () => {
+            assert.strictEqual(Product.db.db.name!, db.name);
+        });
+        it('API ops tests Model.deleteMany()', async () => {
+            const product1 = new Product({name: 'Product 1', price: 10, isCertified: true, category: 'cat 1'});
+            await product1.save();
+            const deleteManyResp = await Product.deleteMany({name: 'Product 1'});
+            assert.strictEqual(deleteManyResp.deletedCount, 1);
+            const findDeletedDoc = await Product.findOne({name: 'Product 1'});
+            assert.strictEqual(findDeletedDoc, null);
+        });
+        it('API ops tests Model.deleteOne()', async () => {
+            const product1 = new Product({name: 'Product 1', price: 10, isCertified: true, category: 'cat 1'});
+            await product1.save();
+            const deleteOneResp = await Product.deleteOne({name: 'Product 1'});
+            assert.strictEqual(deleteOneResp.deletedCount, 1);
+            const findDeletedDoc = await Product.findOne({name: 'Product 1'});
+            assert.strictEqual(findDeletedDoc, null);
+        });
+        it.skip('API ops tests Model.diffIndexes()', async () => {
+            //TODO
+        });
+        //TODO - fix this test
+        it('API ops tests Model.discriminator()', async () => {
+            //Online products have URL
+            const OnlineProduct = Product.discriminator('OnlineProduct', new Schema({url: String}));
+            const regularProduct = new Product({
+                name: 'Product 1',
+                price: 10,
+                isCertified: true,
+                category: 'cat 1',
+                url: 'http://product1.com'
+            });
+            assert.ok(!regularProduct.url);
+            const onlineProduct = new OnlineProduct({
+                name: 'Product 1',
+                price: 10,
+                isCertified: true,
+                category: 'cat 1',
+                url: 'http://product1.com'
+            });
+            assert.ok(onlineProduct.url);
```

Comment:
Would be great to test saving and querying the discriminator model. This test should pass because there's no database operations, but would be good to test that discriminators work. Only potential caveat I can think of with discriminators is that the default discriminator key is `__t`, which should be an acceptable key name in stargate.

Reply:
Added tests to save and read back the discriminator model doc. I could see `__t` getting stored and returned without any issue when I printed the doc while saving and reading.